### PR TITLE
feat(installer): added bun as a js package manager

### PR DESCRIPTION
**Pull request by TheBearodactyl**
# Description

I have added `bun` as a possible `JS` package manager in the install script, and have also prioritized it before `yarn` as a package manager because of the MASSIVE performance increase. 

I have made the output of the `JS` dependency installation prompt use "`NodeJS/BunJS`" instead of `NodeJS` as `BunJS` is independent from `NodeJS`

Along with the output of the `JS` dependency installation prompt, if it uses `bun` as the package manager, it replaces `NodeJS` with `BunJS`

## How Has This Been Tested?

I have run the script both locally and from my forked repo with and without bun installed.

### CODE REVIEW

**Review by cpea2506 — APPROVED** [review]

**Review by Haeven — APPROVED** [review]
LGTM 😄 Nice addition!

**Comment by TheBearodactyl** [comment]
> LGTM 😄 Nice addition!

:D

**Comment by TheBearodactyl** [comment]
the fuck.

**Comment by LostNeophyte** [comment]
lsp template tests don't always work well, you can ignore it

File: utils/installer/install.sh [hunk 1]
```diff
@@ -296,6 +302,8 @@ function __validate_node_installation() {
 
   if [ "$pkg_manager" == "npm" ]; then
     manager_home="$(npm config get prefix 2>/dev/null)"
+  elif [ "$pkg_manager" == "bun" ]; then
+    manager_home="$(echo "$BUN_INSTALL" 2>/dev/null)"
```
Comment (LostNeophyte) [inline utils/installer/install.sh:306]: any reason this can't be just this?
```suggestion
    manager_home="$BUN_INSTALL"
```

**Review by LostNeophyte — CHANGES_REQUESTED** [review]

**Comment by TheBearodactyl** [comment]
> lsp template tests don't always work well, you can ignore it

ah

File: utils/installer/install.sh (hunk 1 above)
Comment (TheBearodactyl) [reply utils/installer/install.sh]: good point

**Review by LostNeophyte — APPROVED** [review]
LGTM, thanks!

**Pull request merged**